src/components/ChatEntry.js

```diff
@@ -1,22 +1,44 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp.js';
+import { useState } from 'react';
+
+
+
 
 const ChatEntry = (props) => {
+  const [likeButton, setLikeButton] = useState('🤍') 
+  const toggleClicked = () => {
+    if (likeButton === '🤍' ) {
+      setLikeButton('❤️')
+      props.updateLikes(true)
+    } else {
+      setLikeButton('🤍');
+      props.updateLikes(false)
+    }
+    
+  }
+  
+  
+
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time"> <TimeStamp time={props.timeStamp}/> </p>
+        <button className="like" onClick= {toggleClicked}>{likeButton}</button>
       </section>
     </div>
   );
 };
 
+// {props.updateLikes}
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
```

Comment:
I would make sure to include `updateLikes` in these `PropTypes` just since it is a large part of making this code work, so leaning on `PropTypes` to catch errors for you is worth the extra typing. (I guess this is what you were planning with the comment on line 37, but I'm still going to point it out.)

I would even go further and mark `updateLikes` as `.isRequired`, though this has the downside of making our test fail due to how we (the instructors) wrote the test data. So fair enough if you did not mark it `.isRequired`.